Jellyfin.Server/Migrations/Routines/MigrateActivityLogDb.cs

```diff
@@ -0,0 +1,125 @@
+
+        /// <inheritdoc/>
+        public string Name => "MigrateActivityLogDatabase";
+
+        /// <inheritdoc/>
+        public void Perform()
+        {
+            var logLevelDictionary = new Dictionary<string, LogLevel>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "None", LogLevel.None },
+                { "Trace", LogLevel.Trace },
+                { "Debug", LogLevel.Debug },
+                { "Information", LogLevel.Information },
+                { "Info", LogLevel.Information },
+                { "Warn", LogLevel.Warning },
+                { "Warning", LogLevel.Warning },
+                { "Error", LogLevel.Error },
+                { "Critical", LogLevel.Critical }
+            };
+
+            var dataPath = _paths.DataPath;
+            using (var connection = SQLite3.Open(
+                Path.Combine(dataPath, DbFilename),
+                ConnectionFlags.ReadOnly,
+                null))
+            {
+                _logger.LogWarning("Migrating the activity database may take a while, do not stop Jellyfin.");
+                using var dbContext = _provider.CreateContext();
+
+                var queryResult = connection.Query("SELECT * FROM ActivityLog ORDER BY Id ASC");
+
+                // Make sure that the database is empty in case of failed migration due to power outages, etc.
+                dbContext.ActivityLogs.RemoveRange(dbContext.ActivityLogs);
+                dbContext.SaveChanges();
+                // Reset the autoincrement counter
+                dbContext.Database.ExecuteSqlRaw("UPDATE sqlite_sequence SET seq = 0 WHERE name = 'ActivityLog';");
+                dbContext.SaveChanges();
+
+                foreach (var entry in queryResult)
+                {
+                    if (!logLevelDictionary.TryGetValue(entry[8].ToString(), out var severity))
+                    {
+                        severity = LogLevel.Trace;
+                    }
+
+                    var newEntry = new ActivityLog(
+                        entry[1].ToString(),
+                        entry[4].ToString(),
+                        entry[6].SQLiteType == SQLiteType.Null ? Guid.Empty : Guid.Parse(entry[6].ToString()))
+                    {
+                        DateCreated = entry[7].ReadDateTime(),
+                        LogSeverity = severity
+                    };
+
+                    if (entry[2].SQLiteType != SQLiteType.Null)
+                    {
+                        newEntry.Overview = entry[2].ToString();
+                    }
+
+                    if (entry[3].SQLiteType != SQLiteType.Null)
+                    {
+                        newEntry.ShortOverview = entry[3].ToString();
+                    }
+
+                    if (entry[5].SQLiteType != SQLiteType.Null)
+                    {
+                        newEntry.ItemId = entry[5].ToString();
+                    }
+
+                    dbContext.ActivityLogs.Add(newEntry);
+                }
```

Comment:
We can also get another performance boost by using `AddRange` instead of calling `Add` within the loop. (NOTE: you'll need to add `using System.Linq;` above to apply this change). The diff on GitHub is long, but I just changed the foreach loop in the first line t use `Select` instead and also last few lines lines to use `AddRange` instead of `Add`

```suggestion
                var newEntries = queryResult.Select(entry =>
                {
                    if (!logLevelDictionary.TryGetValue(entry[8].ToString(), out var severity))
                    {
                        severity = LogLevel.Trace;
                    }

                    var newEntry = new ActivityLog(
                        entry[1].ToString(),
                        entry[4].ToString(),
                        entry[6].SQLiteType == SQLiteType.Null ? Guid.Empty : Guid.Parse(entry[6].ToString()))
                    {
                        DateCreated = entry[7].ReadDateTime(),
                        LogSeverity = severity
                    };

                    if (entry[2].SQLiteType != SQLiteType.Null)
                    {
                        newEntry.Overview = entry[2].ToString();
                    }

                    if (entry[3].SQLiteType != SQLiteType.Null)
                    {
                        newEntry.ShortOverview = entry[3].ToString();
                    }

                    if (entry[5].SQLiteType != SQLiteType.Null)
                    {
                        newEntry.ItemId = entry[5].ToString();
                    }

                    return newEntry;
                });
                
                dbContext.ActivityLogs.AddRange(newEntries);
```